Documentation/contributor-guide/features.md

```diff
@@ -45,6 +46,38 @@ Any new enhancements to the etcd are typically added as an experimental feature.
     - Add a CHANGELOG entry.
 - At least two maintainers must approve feature requirements and related code changes.
 
-### Graduating an Experimental feature to Stable (TODO - spzala)
+### Graduating an Experimental feature to Stable
 
-### Deprecating a feature (TODO - spzala)
+It is important that experimental features don't get stuck in that stage. They should be revisited and moved to the stable stage following the graduation steps as described here.
+
+#### Locate graduation candidate
+Decide if an experimental feature is ready for graduation to the stable stage.
+- Find an issue that was used to enable the experimental feature initially. One way to find such issues is to search for issues with `type/feature` and `experimental` labels.
+- Fix any known open issues against the feature.
+- Make sure the feature was enabled for at least one previous release. Check the PR(s) reference from the issue to see when the feature related code changes were merged.
+
+#### Provide implementation
+If an experimental feature is found ready for graduation to the stable stage, open a Pull Request (PR) with the following changes.
+- Add robust e2e tests if not already provided.
+- Add a new stable feature name identical to the experimental feature name but without the `--experimental` prefix.
+- Deprecate the experimental feature following the [feature deprecation policy](#Deprecating-a-feature).
+- Implementation must ensure that both the graduated and deprecated experimental features work as expected. Note that both these features will co-exist for the timeframe described in the feature deprecation policy.
+- Enable the graduated feature by default if needed.
+
+At least two maintainers must approve the work. Patch releases should not be considered for graduation.
+
+### Deprecating a feature
+
+#### Experimental
+An experimental feature deprecates when it graduates to the stable stage.
```

Comment:
An experimental goes to stable stage or removed directly. 

I don't think there is a deprecation stage for an experimental feature. Of course we can add a warning before removing it.

Reply:
Yes, today we just remove experiment feature when it graduates. In line what we discussed in the previous PR on not surprising users by keeping experimental feature around for one release after it graduate. So, I am proposing deprecation phase to keep features coexist (as we discussed above) with experimental as deprecated. Does that sound good? I agree that experimental feature can get removed directly without being graduated and in that case there is no deprecation. I will make that change. Thanks @ahrtr

Reply:
> So, I am proposing deprecation phase to keep features coexist (as we discussed above) with experimental as deprecated. Does that sound good?

Makes sense to deprecate the experimental flag by adding a comment something like "`Deprecated xxx, use xxx instead`"

Reply:
Thanks @ahrtr !! Also, for the case when experimental feature don't get to graduation, we have already covered it in the previous PR `It can be removed in the next minor or major release without following the feature deprecation policy unless it graduates to the stable future.` under the [Overview section](https://github.com/etcd-io/etcd/blob/bbe848f5c46679a11d30264ddaa10247fcb7a124/Documentation/contributor-guide/features.md#overview)